### DISCUSSION TITLE
Simple sample

### DISCUSSION THREAD

**Discussion by MattPil29** [MediatedCommunications/Mime-Detective#31] [General]
Hi,
I have loaded results by mimetype and now want a quick/simple check for XLSX files.  Is there a way to do this without looping results and looking for a mimetype string?

using MimeDetective;
var Inspector = new ContentInspectorBuilder() {
    Definitions = MimeDetective.Definitions.Default.FileTypes.Documents.All()
}.Build();
var Results = Inspector.Inspect(ContentStream);
var ResultsByMimeType = Results.ByMimeType();

//how to something like this???
if ResultsByMimeType.Contains(defintions.Documents.xlsx)
{ //happy }

Thanks, matt

**Comment by TonyValenti** [comment]
Hi Matt,
While there are a small set of definitions included by default, mind detective is designed to work with the tens of thousands of definitions included in the exhaustive pack.

The right approach is to loop through the results and check either the extensions or the mime types to verify the type of file you were looking for and that it has an appropriate threshold of probability of matching. 

I will likely enhance the read me later today to provide better insight on how the library works.

**Comment by MattPil29** [comment]
Thanks for the quick reply.

I searched your repo and couldn't find the word probability or threshold.  How is that checked and what are good values please?

A little function to check this would be useful to your user base e.g. 

if ResultsByMimeType.ContainsType(defintions.Documents.xlsx, MimeDetective.ProbabilityConfidenceEnum.AverageOrBetter)
{ //happy }

Thanks again, Matthew
